exec1/cache.go

```diff
@@ -0,0 +1,84 @@
+package main
+
+import "container/list"
+
+type Key string
+
+type Cache interface {
+	Set(key Key, value interface{}) bool
+	Get(key Key) (interface{}, bool)
+	Clear()
+}
+
+type LRUCache struct {
+	// количество сохраняемых в кэше элементов
+	cap int
+	// хеш-таблица отображающая ключ (строка) на элемент очереди
+	items map[Key]*list.Element
+	// последних используемых элементов
+	queue *list.List
+}
+
+func NewLRUCache(cap int) *LRUCache {
+
+	return &LRUCache{
+		cap:   cap,
+		items: make(map[Key]*list.Element, cap),
+		queue: list.New(),
+	}
+}
+
+func (LRU *LRUCache) Set(key Key, value interface{}) bool {
+
+	el, ok := LRU.items[key]
+	newEl := LRU.queue.PushFront(value)
+	LRU.items[key] = newEl
+
+	if ok {
+
+		LRU.queue.Remove(el)
+
+		return true
+	}
+
+	if LRU.queue.Len() > LRU.cap {
```

Comment:
Уменьшим так вложенность, а если она равна, то удаляем последний элемент

```suggestion
	if LRU.queue.Len() < LRU.cap {
	    return false
	}
```

Reply:
+